week-07/day-04/HelloDIProject/HelloDIProject/Controllers/UsefulController.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using HelloDIProject;
+
+// For more information on enabling MVC for empty projects, visit https://go.microsoft.com/fwlink/?LinkID=397860
+
+namespace HelloDIProject.Controllers
+{
+    [Route("useful")]
+    public class UsefulController : Controller
+    {
+
+        UtilityService utilityService;
+
+        public UsefulController (UtilityService utilityService)
+        {
+            this.utilityService = utilityService;
+        }
+
+        [Route("")]
+        public IActionResult Useful()
+        {
+            return View();
+        }
+
+        // GET: /<controller>/
+        [Route("colored")]
+        public IActionResult Colored()
+        {
+            return View(utilityService);
+        }
+
+        [Route("email")]
+        public IActionResult Email(String email, UtilityService utilityService)
```

Comment:
I don't see how the utilityService can end up here as second argument. And if it works, it shouldn't. If this is a `GET` route handling method, all the input parameters are coming from a URL. Contents of a whole service shouldn't be there.

e. g.:

`/email?email=martin.malinda@greenfoxacademy.com`